src/main/java/org/wildfly/security/sasl/util/AbstractSaslClient.java

```diff
@@ -42,8 +44,8 @@
      * @param authorizationId the possibly {@code null} protocol-dependent name used for authorization
      * @param hasInitialResponse {@code true} if the mechanism supports an initial response, {@code false} otherwise
      */
-    protected AbstractSaslClient(final String mechanismName, final String protocol, final String serverName, final CallbackHandler callbackHandler, final String authorizationId, final boolean hasInitialResponse) {
-        super(mechanismName, protocol, serverName, callbackHandler);
+    protected AbstractSaslClient(final String mechanismName, final String protocol, final String serverName, final CallbackHandler callbackHandler, final String authorizationId, final boolean hasInitialResponse, ElytronMessages log) {
```

Comment:
This is where things start to get a little more problematic, this class has been flagged as being in a public API package - additionally this class is public so is visible outside of this package.

In retrospect maybe we didn't want to have a common base others can extend but unfortunately we do have that today.

So the bad news is we can't just change the existing method signature of the constructor.

One option is to overload the constructor, if the form without the logger is used it can call the form that takes the logger and use one of the existing categories.  Alternatively the abstract class could add a ElytronMessages getLogger() method that by default returns a common logger and the base types override this method to return the correct logger.